x/interchainqueries/keeper/keeper.go

```diff
@@ -244,17 +252,21 @@ func (k Keeper) updateLastLocalHeight(ctx sdk.Context, query *types.RegisteredQu
 }
 
 // checkLastRemoteHeight checks whether the given height is greater than the query's remote height
-// of the last result submission.
-func (k Keeper) checkLastRemoteHeight(ctx sdk.Context, query types.RegisteredQuery, height uint64) error {
-	if query.LastSubmittedResultRemoteHeight >= height {
+// of the last result submission. Also it checks if revision number has updated. If yes, we don't care about revision height
+// see
+func (k Keeper) checkLastRemoteHeight(ctx sdk.Context, query types.RegisteredQuery, height ibcclienttypes.Height) error {
+	if query.LastSubmittedResultRemoteHeight.RevisionNumber < height.GetRevisionNumber() {
+		return nil
+	} else if query.LastSubmittedResultRemoteHeight.RevisionHeight >= height.GetRevisionHeight() &&
+		query.LastSubmittedResultRemoteHeight.RevisionNumber == height.GetRevisionNumber() {
```

Comment:
`ibcclienttypes.Height` has all the necessary methods to compare two structs (`LTE`, `GTE`, etc).